File: vecvsfor.js
#!/usr/local/bin/node

var VEC_LEN = 1000000
var AVERAGING = 1000

var A = new Array(VEC_LEN);
var B = new Array(VEC_LEN);

for (var i = 0; i < VEC_LEN; ++i) {
  A[i] = Math.random();
  B[i] = Math.random();
}

var c = 0.0;

var start = new Date().getTime();

for (var j = 0; j < AVERAGING; ++j) {
  c = 0;
  for (var i = 0; i < VEC_LEN; ++i) {
    c += A[i] * B[i];
  }
}

var elapsed = new Date().getTime() - start;

console.log("c:", c);
console.log("For loop:", elapsed / AVERAGING);

File: vecvsfor.py
#!/usr/local/bin/python3

import numpy as np
import time

VEC_LEN = 1000000
AVERAGING = 1000

a = np.random.rand(VEC_LEN)
b = np.random.rand(VEC_LEN)

tic = time.time()
for i in range(AVERAGING):
  c = np.dot(a,b)
toc = time.time()

print("c: " + str(c))
print("Vectorized version: " + str ((toc - tic) * 1000 / AVERAGING) + "ms")

c = 0

tic = time.time()
for i in range(VEC_LEN):
  c += a[i]* b[i]
toc = time.time()

print("c: " + str(c))
print("For loop: " + str ((toc-tic) * 1000) + "ms")
